arch/risc-v/src/mpfs/mpfs_opensbi.c

```diff
@@ -23,27 +23,36 @@
  ****************************************************************************/
 
 #include <nuttx/config.h>
-
 #include <assert.h>
 #include <errno.h>
 #include <stdint.h>
-
-#include "riscv_internal.h"
-#include "riscv_arch.h"
+#include <riscv_arch.h>
 
 #include <hardware/mpfs_plic.h>
 #include <hardware/mpfs_memorymap.h>
 #include <hardware/mpfs_clint.h>
 #include <hardware/mpfs_sysreg.h>
 
+/* OpenSBI will also define NULL. Undefine NULL in order to avoid warning:
+ * 'warning: "NULL" redefined'
+ */
+
+#ifdef NULL
+  #undef NULL
+#endif
+
 #include <sbi/sbi_types.h>
 #include <sbi/riscv_atomic.h>
+#include <sbi/riscv_asm.h>
```

Comment:
please check, but I think it is not needed here

Reply:
true